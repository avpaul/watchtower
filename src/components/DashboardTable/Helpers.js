const getColumnAttribute = headerName => {
  switch (headerName) {
    case 'Level':
    case 'Communication':
    case 'Initiative':
    case 'Integration':
    case 'Professionalism':
    case 'Quality':
    case 'Quantity':
      return headerName.toLowerCase();
    case 'Fellow Name':
      return 'name';
    case 'Week':
      return 'weeksSpent';
    case 'LF/TTL':
      return 'managerName';
    case 'DevPulse Status':
      return 'devPulseStatus';
    case 'LMS Status':
      return 'lmsStatus';
    case 'Advancement':
      return 'overall_status';
    case 'Status':
      return 'lmsStatus';
    case 'LMS Outputs':
      return 'submitted';
    case 'LMS Outputs > 2':
      return 'satisfied';
    default:
      return 'name';
  }
};

export default getColumnAttribute;
