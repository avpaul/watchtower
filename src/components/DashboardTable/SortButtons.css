.sort-buttons {
  min-width: 10px;
  min-height: 5px;
}
.sort-buttons i {
  width: 5px;
  font-size: 10px;
  cursor: pointer;
}
.sort-buttons i:focus,
.sort-buttons i:hover,
.sort-buttons i:active {
  outline: none;
  border: inherit;
  cursor: pointer;
}

.sort-buttons div {
  height: 10px;
  width: 15px;
  text-align: center;
}
