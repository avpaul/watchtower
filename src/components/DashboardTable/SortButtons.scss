.sort-buttons {
  min-width: 10px;
  min-height: 5px;
  width: 15px;
  height: 15px;
  position: relative;
}

.sort-buttons i {
  width: 5px;
  font-size: 10px;
  cursor: pointer;
}

.sort-buttons i:focus,
.sort-buttons i:hover,
.sort-buttons i:active {
  outline: none;
  border: inherit;
  cursor: pointer;
}

.sort-buttons__up,
.sort-buttons__down {
  width: 100%;
  height: 10px;
  position: absolute;
  right: 0;
}

.sort-buttons__up {
  top: 0;
}

.sort-buttons__down {
  bottom: 0;
}
