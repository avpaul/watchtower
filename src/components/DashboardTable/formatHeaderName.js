const formatHeaderName = headerName => {
  let formatedHeaderName;
  switch (headerName) {
    case 'Fellow Name':
      formatedHeaderName = 'firstName';
      break;
    case 'Level':
      formatedHeaderName = 'level';
      break;
    case 'Week':
      formatedHeaderName = 'weeksSpent';
      break;
    case 'LF/TTL':
      formatedHeaderName = 'submitterFirstName';
      break;
    case 'DevPulse Status':
      formatedHeaderName = 'devPulseStatus';
      break;
    case 'LMS Status':
      formatedHeaderName = 'lmsStatus';
      break;
    case 'Advancement':
      formatedHeaderName = 'advanceStatus';
      break;
    case 'Communication':
      formatedHeaderName = 'communication';
      break;
    case 'Initiative':
      formatedHeaderName = 'initiative';
      break;
    case 'Integration':
      formatedHeaderName = 'integration';
      break;
    case 'Professionalism':
      formatedHeaderName = 'professionalism';
      break;
    case 'Quality':
      formatedHeaderName = 'quality';
      break;
    case 'Quantity':
      formatedHeaderName = 'quantity';
      break;
    case 'Status':
      formatedHeaderName = 'lmsStatus';
      break;
    case 'LMS Outputs':
      formatedHeaderName = 'lmsOutputs';
      break;
    case 'LMS Outputs > 2':
      formatedHeaderName = 'lmsOutput';
      break;
    default:
      formatedHeaderName = 'firstName';
  }
  return formatedHeaderName;
};

export default formatHeaderName;
