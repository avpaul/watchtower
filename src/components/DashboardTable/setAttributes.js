const ratingsAttributes = [
  'quality',
  'quantity',
  'initiative',
  'communication',
  'professionalism',
  'integration',
  'devPulseStatus',
  'lmsStatus',
  'weeksSpent',
  'advanceStatus'
];
const setColor = (element, fellow) => {
  let attrColor;
  const value = fellow[element];
  if (
    value < 1 ||
    value === 'Off Track' ||
    value === 'On PIP'
  ) {
    attrColor = 'off-track';
  } else if (value === 'N/A') {
    attrColor = 'no-track';
  }
  return attrColor;
}

const cellRatingsStatusAttr = (element, fellow) => {
  let attrColor;
  const value = fellow[element];
  if (ratingsAttributes.includes(element)) {
    attrColor = setColor(element, fellow);
  }
  return { key: element, value, color: attrColor };
}

const cellAttr = (element, fellow) => {
  const value = fellow[element];
  let fellowCell;
  if (element === 'name') {
    fellowCell = { key: 'name', value: `${fellow.firstName} ${fellow.lastName}` };
  } else if (element === 'ttlName') {
    fellowCell = {
      key: 'ttlName',
      value: `${fellow.submitterFirstName} ${fellow.submitterLastName}`
    };
  } else if (value === undefined) {
    fellowCell = { key: element, value: 'N/A', color: 'off-track' };
  } else {
    fellowCell = cellRatingsStatusAttr(element, fellow);
  }
  return fellowCell;
}

export default cellAttr;
