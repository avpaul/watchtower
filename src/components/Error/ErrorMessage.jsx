import React from 'react';
import PropTypes from 'prop-types';

const ErrorMessage = ({ message, customStyle }) => {
  const defaultStyle = {
    width: '100%',
    margin: '1rem auto',
    fontSize: '2rem',
    textAlign: 'center',
  };

  const errorMessageStyle = { ...defaultStyle, ...customStyle };
  return (
    <div style={errorMessageStyle}>{message}</div>
  );
};

ErrorMessage.defaultProps = {
  customStyle: {},
};

ErrorMessage.propTypes = {
  message: PropTypes.string.isRequired,
  customStyle: PropTypes.objectOf(PropTypes.string),
};

export default ErrorMessage;
