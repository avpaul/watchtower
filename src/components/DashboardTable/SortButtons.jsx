import React from 'react';
import PropTypes from 'prop-types';
import './SortButtons.scss';

const SortButtons = ({
  arrowUpClick,
  arrowDownClick,
  headerName,
  active,
  sortType
}) => {
  const arrowClass = 'fas align-self-center';
  const arrowUpClass = `${arrowClass} fa-angle-up ${
    active && sortType === 'ascending' ? 'text-primary' : ''
  }`;
  const arrowDownClass = `${arrowClass} fa-angle-down ${
    active && sortType === 'descending' ? 'text-primary' : ''
  }`;

  return (
    <span className="sort-buttons ml-3 p-0 text-left">
      <div className="sort-buttons__up p-0">
        <i
          className={arrowUpClass}
          onClick={arrowUpClick}
          onKeyPress={arrowUpClick}
          role="button"
          tabIndex="0"
          data-target={headerName}
          data-ascending
        />
      </div>
      <div className="sort-buttons__down p-0">
        <i
          className={arrowDownClass}
          onClick={arrowDownClick}
          onKeyPress={arrowDownClick}
          role="button"
          tabIndex="-1"
          data-target={headerName}
        />
      </div>
    </span>
  );
};

SortButtons.propTypes = {
  arrowUpClick: PropTypes.func.isRequired,
  arrowDownClick: PropTypes.func.isRequired,
  headerName: PropTypes.string.isRequired,
  active: PropTypes.bool.isRequired,
  sortType: PropTypes.string.isRequired
};

export default SortButtons;
