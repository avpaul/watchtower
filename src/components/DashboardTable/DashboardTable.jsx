import React, { Fragment, Component } from 'react';
import PropTypes from 'prop-types';
import Table from '../TableComponents/Table';
import Cell from '../TableComponents/Cell';
import Row from '../TableComponents/Row';
import DashboardRow from './DashboardRow';
import Error from '../Error';
import Loader from '../Loader/Loader';
import cellAttr from './setAttributes';
import formatHeaderName from './formatHeaderName';
import SortButtons from './SortButtons';

const { ErrorMessage } = Error;

class DashboardTable extends Component {
  constructor(props) {
    super(props);
    this.state = {
      sortBy: 'firstName',
      sortType: 'ascending'
    };
  }

  sortFellows = (list, name, sortType) => {
    if (sortType === 'descending') {
      const sortedList = list.sort((a, b) => {
        const A = typeof a[name] === 'string' ? a[name].toLowerCase() : a[name];
        const B = typeof b[name] === 'string' ? b[name].toLowerCase() : b[name];
        if (A < B) return 1;
        if (A > B) return -1;
        return 0;
      });
      return sortedList;
    }
    if (sortType === 'ascending') {
      const sortedList = list.sort((a, b) => {
        const A = typeof a[name] === 'string' ? a[name].toLowerCase() : a[name];
        const B = typeof b[name] === 'string' ? b[name].toLowerCase() : b[name];

        if (A < B) return -1;
        if (A > B) return 1;
        return 0;
      });
      return sortedList;
    }
    return list;
  };

  fellowCells = fellow => {
    const { cellValues } = this.props;
    return cellValues.map(element => cellAttr(element, fellow));
  };

  arrowUpClick = event => {
    const name = event.target.getAttribute('data-target');
    this.setState({ sortBy: name, sortType: 'descending' });
  };

  arrowDownClick = event => {
    const name = event.target.getAttribute('data-target');
    this.setState({ sortBy: name, sortType: 'ascending' });
  };

  render() {
    const { fellows, loading, headers } = this.props;
    const { sortBy, sortType } = this.state;
    if (fellows.length < 1 && !loading) {
      return (
        <ErrorMessage message="There's currently no fellows matching the filter and/or search." />
      );
    }

    return (
      <Fragment>
        <Table>
          <Row header>
            {headers.map(element => {
              let active = false;
              const headerName = formatHeaderName(element);
              if (headerName.toLowerCase() === sortBy.toLowerCase()) {
                active = true;
              }
              return (
                <Cell>
                  <span className="row">
                    <span className="pl-2 pt-2">{element}</span>
                    <div className="">
                      <SortButtons
                        arrowUpClick={this.arrowUpClick}
                        arrowDownClick={this.arrowDownClick}
                        handleCardClick={this.handleCardClick}
                        headerName={headerName}
                        active={active}
                        sortType={sortType}
                      />
                    </div>
                  </span>
                </Cell>
              );
            })}
          </Row>

          {this.sortFellows(fellows, sortBy, sortType).map(fellow => (
            <DashboardRow
              key={fellow.id}
              fellow={fellow}
              fellowCells={this.fellowCells(fellow)}
            />
          ))}
        </Table>
        {loading && <Loader />}
      </Fragment>
    );
  }
}

DashboardTable.propTypes = {
  fellows: PropTypes.arrayOf(PropTypes.object).isRequired,
  loading: PropTypes.bool.isRequired,
  headers: PropTypes.arrayOf(PropTypes.string).isRequired,
  cellValues: PropTypes.arrayOf(PropTypes.string).isRequired
};

export default DashboardTable;
