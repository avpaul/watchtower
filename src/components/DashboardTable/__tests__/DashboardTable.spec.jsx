import React from 'react';
import { shallow } from 'enzyme';

import DashboardTable from '../DashboardTable';
import fellows from '../../../__mocks__/fellows';
import ErrorMessage from '../../Error/ErrorMessage';
import table from '../../../views/DashboardPage/tableHeaders';
import SortButtons from '../SortButtons';

describe('Dashboard Table ', () => {
  const props = {
    fellows,
    fellowsToDisplay: fellows,
    loading: false,
    headers: table.default.titles,
    cellValues: table.default.cells,
    handleSortingChange: jest.fn()
  };

  const getEvent = isAscending => ({
    target: {
      getAttribute: attr => {
        switch (attr) {
          case 'data-target':
            return 'name';
          case 'data-ascending':
            return isAscending;
          default:
            return '';
        }
      }
    }
  });

  it('renders ErrorMessage when no fellows and is not fetching fellows', () => {
    const newProps = { ...props, fellows: [], fellowsToDisplay: [] };
    const wrapper = shallow(<DashboardTable {...newProps} />);

    expect(
      wrapper.contains(
        <ErrorMessage message="There's currently no fellows matching the filter and/or search." />
      )
    ).toEqual(true);
  });

  it('renders table cells in the correct order', () => {
    const arrowUpEvent = getEvent(true);
    const arrowDownEvent = getEvent(false);

    const wrapper = shallow(<DashboardTable {...props} />);
    expect(wrapper).toMatchSnapshot();
    wrapper.instance().arrowClick(arrowUpEvent);
    expect(wrapper.state('sortBy')).toBe('name');
    wrapper.instance().arrowClick(arrowDownEvent);
    expect(wrapper.state('sortType')).toBe('descending');
  });

  it('renders the sort buttons to match snapshot', () => {
    const wrapper = shallow(
      <SortButtons
        arrowUpClick={jest.fn()}
        arrowDownClick={jest.fn()}
        handleCardClick={jest.fn()}
        headerName="header"
        active
        sortType="ascending"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders table cell values', () => {
    const wrapper = shallow(<DashboardTable {...props} />);
    wrapper.setState({ sortBy: 'weeksSpent' });
    wrapper.instance().sortFellows();
    expect(wrapper.state('sortBy')).toBe('weeksSpent');
  });
});
