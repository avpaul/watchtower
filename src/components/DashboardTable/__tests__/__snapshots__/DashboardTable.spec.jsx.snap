// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Table  renders table cells in the correct order 1`] = `
<Fragment>
  <Table>
    <Row
      header={true}
    >
      <Cell
        addedClass=""
        key="weak-key-0"
        title=""
      >
        <span
          className="row"
        >
          <span
            className="pl-2 pt-2"
          >
            Fellow Name
          </span>
          <div>
            <SortButtons
              active={true}
              arrowDownClick={[Function]}
              arrowUpClick={[Function]}
              headerName="name"
              sortType="ascending"
            />
          </div>
        </span>
      </Cell>
      <Cell
        addedClass=""
        key="weak-key-1"
        title=""
      >
        <span
          className="row"
        >
          <span
            className="pl-2 pt-2"
          >
            Level
          </span>
          <div>
            <SortButtons
              active={false}
              arrowDownClick={[Function]}
              arrowUpClick={[Function]}
              headerName="level"
              sortType="ascending"
            />
          </div>
        </span>
      </Cell>
      <Cell
        addedClass=""
        key="weak-key-2"
        title=""
      >
        <span
          className="row"
        >
          <span
            className="pl-2 pt-2"
          >
            Week
          </span>
          <div>
            <SortButtons
              active={false}
              arrowDownClick={[Function]}
              arrowUpClick={[Function]}
              headerName="weeksSpent"
              sortType="ascending"
            />
          </div>
        </span>
      </Cell>
      <Cell
        addedClass=""
        key="weak-key-3"
        title=""
      >
        <span
          className="row"
        >
          <span
            className="pl-2 pt-2"
          >
            LF/TTL
          </span>
          <div>
            <SortButtons
              active={false}
              arrowDownClick={[Function]}
              arrowUpClick={[Function]}
              headerName="managerName"
              sortType="ascending"
            />
          </div>
        </span>
      </Cell>
      <Cell
        addedClass=""
        key="weak-key-4"
        title=""
      >
        <span
          className="row"
        >
          <span
            className="pl-2 pt-2"
          >
            DevPulse Status
          </span>
          <div>
            <SortButtons
              active={false}
              arrowDownClick={[Function]}
              arrowUpClick={[Function]}
              headerName="devPulseStatus"
              sortType="ascending"
            />
          </div>
        </span>
      </Cell>
      <Cell
        addedClass=""
        key="weak-key-5"
        title=""
      >
        <span
          className="row"
        >
          <span
            className="pl-2 pt-2"
          >
            LMS Status
          </span>
          <div>
            <SortButtons
              active={false}
              arrowDownClick={[Function]}
              arrowUpClick={[Function]}
              headerName="lmsStatus"
              sortType="ascending"
            />
          </div>
        </span>
      </Cell>
      <Cell
        addedClass=""
        key="weak-key-6"
        title=""
      >
        <span
          className="row"
        >
          <span
            className="pl-2 pt-2"
          >
            Advancement
          </span>
          <div>
            <SortButtons
              active={false}
              arrowDownClick={[Function]}
              arrowUpClick={[Function]}
              headerName="overall_status"
              sortType="ascending"
            />
          </div>
        </span>
      </Cell>
    </Row>
    <DashboardRow
      fellow={
        Object {
          "actual apprenticeship start date": "2018-12-10",
          "advanceStatus": "N/A",
          "cohort": "Class 1 - NBO",
          "communication": "1.00",
          "devPulseStatus": "On Track",
          "email": "abbey.maddy@andela.com",
          "fellow_id": "-LMKN5flPv4R7LCZs0lh",
          "firstName": "Abbey",
          "initiative": "1.00",
          "integration": "1.00",
          "isBeforeWk5": true,
          "lastName": "Maddy",
          "level": "D0A Simulations",
          "lmsStatus": "N/A",
          "professionalism": "1.00",
          "quality": "1.00",
          "quantity": "1.00",
          "simulation start date": "2018-11-01",
          "staffId": null,
          "status": "onTrack",
          "submitterFirstName": "",
          "submitterLastName": "",
          "weeksSpent": 23,
        }
      }
      fellowCells={
        Array [
          Object {
            "color": "no-track",
            "element": "name",
            "value": "N/A",
          },
          Object {
            "color": null,
            "element": "level",
            "value": "D0A Simulations",
          },
          Object {
            "color": null,
            "element": "weeksSpent",
            "value": 23,
          },
          Object {
            "color": "no-track",
            "element": "ttlName",
            "value": "N/A",
          },
          Object {
            "color": null,
            "element": "devPulseStatus",
            "value": "On Track",
          },
          Object {
            "color": "no-track",
            "element": "lmsStatus",
            "value": "N/A",
          },
          Object {
            "color": "no-track",
            "element": "advanceStatus",
            "value": "N/A",
          },
        ]
      }
      key="-LMKN5flPv4R7LCZs0lh"
    />
    <DashboardRow
      fellow={
        Object {
          "advanceStatus": "N/A",
          "cohort": "Class 10 - KLA",
          "communication": "1.12",
          "devPulseStatus": "Off Track",
          "email": "gordon.balaaka@andela.com",
          "fellow_id": "-LJK1qECFdekucrwqqkh",
          "firstName": "Gordon",
          "initiative": "1.00",
          "integration": "1.25",
          "isBeforeWk5": false,
          "lastName": "Balaaka",
          "level": "D0A Simulations",
          "levelReturned": "D0A Simulations",
          "lmsStatus": "N/A",
          "professionalism": "1.00",
          "quality": "1.00",
          "quantity": "1.12",
          "staffId": "-LAmk78f7GyBWkt2UWpb",
          "status": "gteWk5OffTrack",
          "submitterFirstName": "Ephraim",
          "submitterLastName": "Malinga",
          "weeksSpent": "N/A",
        }
      }
      fellowCells={
        Array [
          Object {
            "color": "no-track",
            "element": "name",
            "value": "N/A",
          },
          Object {
            "color": null,
            "element": "level",
            "value": "D0A Simulations",
          },
          Object {
            "color": "no-track",
            "element": "weeksSpent",
            "value": "N/A",
          },
          Object {
            "color": "no-track",
            "element": "ttlName",
            "value": "N/A",
          },
          Object {
            "color": "off-track",
            "element": "devPulseStatus",
            "value": "Off Track",
          },
          Object {
            "color": "no-track",
            "element": "lmsStatus",
            "value": "N/A",
          },
          Object {
            "color": "no-track",
            "element": "advanceStatus",
            "value": "N/A",
          },
        ]
      }
      key="-LJK1qECFdekucrwqqkh"
    />
    <DashboardRow
      fellow={
        Object {
          "actual apprenticeship start date": "2018-10-15",
          "advanceStatus": "N/A",
          "cohort": "Class 13 - KLA",
          "communication": "1.00",
          "devPulseStatus": "On Track",
          "email": "timothy.mubiru@andela.com",
          "expected apprenticeship completion date": "2019-01-16",
          "fellow_id": "-LJK2KDhq70_X1YbirFx",
          "firstName": "Timothy",
          "initiative": "1.00",
          "integration": "1.00",
          "isBeforeWk5": true,
          "lastName": "Mubiru",
          "level": "D0A Simulations",
          "lmsStatus": "N/A",
          "professionalism": "1.00",
          "quality": "1.00",
          "quantity": "1.00",
          "staffId": "-LAmk78f7GyBWkt2UWpb",
          "status": "onTrack",
          "submitterFirstName": "Ephraim",
          "submitterLastName": "Malinga",
          "weeksSpent": "N/A",
        }
      }
      fellowCells={
        Array [
          Object {
            "color": "no-track",
            "element": "name",
            "value": "N/A",
          },
          Object {
            "color": null,
            "element": "level",
            "value": "D0A Simulations",
          },
          Object {
            "color": "no-track",
            "element": "weeksSpent",
            "value": "N/A",
          },
          Object {
            "color": "no-track",
            "element": "ttlName",
            "value": "N/A",
          },
          Object {
            "color": null,
            "element": "devPulseStatus",
            "value": "On Track",
          },
          Object {
            "color": "no-track",
            "element": "lmsStatus",
            "value": "N/A",
          },
          Object {
            "color": "no-track",
            "element": "advanceStatus",
            "value": "N/A",
          },
        ]
      }
      key="-LJK2KDhq70_X1YbirFx"
    />
    <DashboardRow
      fellow={
        Object {
          "actual apprenticeship start date": "2019-02-18",
          "actual simulations completion date": "2019-02-14",
          "advanceStatus": "N/A",
          "cohort": "Class 12 - KLA",
          "communication": "1.00",
          "devPulseStatus": "On Track",
          "email": "sam.rubarema@andela.com",
          "expected apprenticeship completion date": "2019-05-16",
          "fellow_id": "-LP641ReDxZeT2ABACW8",
          "firstName": "Sam",
          "initiative": "1.00",
          "integration": "1.00",
          "isBeforeWk5": true,
          "lastName": "Rubarema",
          "level": "D0B Apprenticeship",
          "lmsStatus": "N/A",
          "professionalism": "1.00",
          "quality": "1.00",
          "quantity": "1.00",
          "simulation start date": "2018-11-19",
          "staffId": null,
          "status": "onTrack",
          "submitterFirstName": "",
          "submitterLastName": "",
          "weeksSpent": 7,
        }
      }
      fellowCells={
        Array [
          Object {
            "color": "no-track",
            "element": "name",
            "value": "N/A",
          },
          Object {
            "color": null,
            "element": "level",
            "value": "D0B Apprenticeship",
          },
          Object {
            "color": null,
            "element": "weeksSpent",
            "value": 7,
          },
          Object {
            "color": "no-track",
            "element": "ttlName",
            "value": "N/A",
          },
          Object {
            "color": null,
            "element": "devPulseStatus",
            "value": "On Track",
          },
          Object {
            "color": "no-track",
            "element": "lmsStatus",
            "value": "N/A",
          },
          Object {
            "color": "no-track",
            "element": "advanceStatus",
            "value": "N/A",
          },
        ]
      }
      key="-LP641ReDxZeT2ABACW8"
    />
    <DashboardRow
      fellow={
        Object {
          "actual apprenticeship start date": "2019-01-14",
          "actual simulations completion date": "2019-01-11",
          "advanceStatus": "N/A",
          "cohort": "Class 41 - LOS",
          "communication": "1.00",
          "devPulseStatus": "On Track",
          "email": "fredrick.adegoke@andela.com",
          "expected apprenticeship completion date": "2019-04-08",
          "fellow_id": "-LLUW2BwQiZS0j9KF10j",
          "firstName": "Fredrick",
          "initiative": "1.00",
          "integration": "1.00",
          "isBeforeWk5": true,
          "lastName": "Adegoke",
          "level": "D0B Apprenticeship",
          "lmsStatus": "N/A",
          "professionalism": "1.00",
          "quality": "1.00",
          "quantity": "1.00",
          "simulation start date": "2018-10-16",
          "staffId": "-LGy4OuPDHCZCZvDuPz0",
          "status": "onTrack",
          "submitterFirstName": "Trust",
          "submitterLastName": "Birungi",
          "weeksSpent": 12,
        }
      }
      fellowCells={
        Array [
          Object {
            "color": "no-track",
            "element": "name",
            "value": "N/A",
          },
          Object {
            "color": null,
            "element": "level",
            "value": "D0B Apprenticeship",
          },
          Object {
            "color": null,
            "element": "weeksSpent",
            "value": 12,
          },
          Object {
            "color": "no-track",
            "element": "ttlName",
            "value": "N/A",
          },
          Object {
            "color": null,
            "element": "devPulseStatus",
            "value": "On Track",
          },
          Object {
            "color": "no-track",
            "element": "lmsStatus",
            "value": "N/A",
          },
          Object {
            "color": "no-track",
            "element": "advanceStatus",
            "value": "N/A",
          },
        ]
      }
      key="-LLUW2BwQiZS0j9KF10j"
    />
    <DashboardRow
      fellow={
        Object {
          "advanceStatus": "N/A",
          "cohort": "Class 38 - LOS",
          "communication": "1.00",
          "devPulseStatus": "Off Track",
          "email": "sinmiloluwa.oloyede@andela.com",
          "fellow_id": "-LEnW5RRn0IkO8VjnMma",
          "firstName": "Sinmiloluwa",
          "initiative": "1.20",
          "integration": "1.10",
          "isBeforeWk5": false,
          "lastName": "Oloyede",
          "level": "D0B Apprenticeship",
          "levelReturned": "D0B Apprenticeship",
          "lmsStatus": "N/A",
          "professionalism": "1.10",
          "quality": "1.10",
          "quantity": "1.30",
          "staffId": "-LGy4OuPDHCZCZvDuPz0",
          "status": "gteWk5OffTrack",
          "submitterFirstName": "Trust",
          "submitterLastName": "Birungi",
          "weeksSpent": "N/A",
        }
      }
      fellowCells={
        Array [
          Object {
            "color": "no-track",
            "element": "name",
            "value": "N/A",
          },
          Object {
            "color": null,
            "element": "level",
            "value": "D0B Apprenticeship",
          },
          Object {
            "color": "no-track",
            "element": "weeksSpent",
            "value": "N/A",
          },
          Object {
            "color": "no-track",
            "element": "ttlName",
            "value": "N/A",
          },
          Object {
            "color": "off-track",
            "element": "devPulseStatus",
            "value": "Off Track",
          },
          Object {
            "color": "no-track",
            "element": "lmsStatus",
            "value": "N/A",
          },
          Object {
            "color": "no-track",
            "element": "advanceStatus",
            "value": "N/A",
          },
        ]
      }
      key="-LEnW5RRn0IkO8VjnMma"
    />
    <DashboardRow
      fellow={
        Object {
          "advanceStatus": "N/A",
          "cohort": "Class 10 - KLA",
          "communication": "1.25",
          "devPulseStatus": "Off Track",
          "email": "samuel.bwambale@andela.com",
          "fellow_id": "-LJK2fj_k_R7aYBAq_AR",
          "firstName": "Samuel",
          "initiative": "1.25",
          "integration": "1.50",
          "isBeforeWk5": false,
          "lastName": "Bwambale",
          "level": "D0A Simulations",
          "levelReturned": "D0A Simulations",
          "lmsStatus": "N/A",
          "professionalism": "1.50",
          "quality": "1.00",
          "quantity": "1.12",
          "staffId": "-LJXG90epIwmW2U1XQHB",
          "status": "gteWk5OffTrack",
          "submitterFirstName": "Martin",
          "submitterLastName": "Bbaale",
          "weeksSpent": "N/A",
        }
      }
      fellowCells={
        Array [
          Object {
            "color": "no-track",
            "element": "name",
            "value": "N/A",
          },
          Object {
            "color": null,
            "element": "level",
            "value": "D0A Simulations",
          },
          Object {
            "color": "no-track",
            "element": "weeksSpent",
            "value": "N/A",
          },
          Object {
            "color": "no-track",
            "element": "ttlName",
            "value": "N/A",
          },
          Object {
            "color": "off-track",
            "element": "devPulseStatus",
            "value": "Off Track",
          },
          Object {
            "color": "no-track",
            "element": "lmsStatus",
            "value": "N/A",
          },
          Object {
            "color": "no-track",
            "element": "advanceStatus",
            "value": "N/A",
          },
        ]
      }
      key="-LJK2fj_k_R7aYBAq_AR"
    />
    <DashboardRow
      fellow={
        Object {
          "advanceStatus": "N/A",
          "cohort": "Class 37 - LOS",
          "communication": "0.00",
          "devPulseStatus": "Off Track",
          "email": "folajimi.ogunbadejo@andela.com",
          "fellow_id": "-LD3PA3VgP4qvRRORX08",
          "firstName": "Folajimi",
          "initiative": "1.33",
          "integration": "1.00",
          "isBeforeWk5": false,
          "lastName": "Ogunbadejo",
          "level": "D0B Apprenticeship",
          "levelReturned": "D0B Apprenticeship",
          "lmsStatus": "N/A",
          "professionalism": "0.00",
          "quality": "1.00",
          "quantity": "1.41",
          "staffId": "-LGy4OuPDHCZCZvDuPz0",
          "status": "gteWk5OffTrack",
          "submitterFirstName": "Trust",
          "submitterLastName": "Birungi",
          "weeksSpent": "N/A",
        }
      }
      fellowCells={
        Array [
          Object {
            "color": "no-track",
            "element": "name",
            "value": "N/A",
          },
          Object {
            "color": null,
            "element": "level",
            "value": "D0B Apprenticeship",
          },
          Object {
            "color": "no-track",
            "element": "weeksSpent",
            "value": "N/A",
          },
          Object {
            "color": "no-track",
            "element": "ttlName",
            "value": "N/A",
          },
          Object {
            "color": "off-track",
            "element": "devPulseStatus",
            "value": "Off Track",
          },
          Object {
            "color": "no-track",
            "element": "lmsStatus",
            "value": "N/A",
          },
          Object {
            "color": "no-track",
            "element": "advanceStatus",
            "value": "N/A",
          },
        ]
      }
      key="-LD3PA3VgP4qvRRORX08"
    />
    <DashboardRow
      fellow={
        Object {
          "advanceStatus": "N/A",
          "cohort": "Class 10 - KLA",
          "communication": "1.33",
          "devPulseStatus": "Off Track",
          "email": "edward.kigozi@andela.com",
          "fellow_id": "-LJK2KDhq70_X1YbirFa",
          "firstName": "Edward",
          "initiative": "1.33",
          "integration": "1.33",
          "isBeforeWk5": true,
          "lastName": "Kigozi",
          "level": "D0A Simulations",
          "levelReturned": "D0A Simulations",
          "lmsStatus": "N/A",
          "professionalism": "1.33",
          "quality": "1.33",
          "quantity": "1.33",
          "staffId": "-LAmk78f7GyBWkt2UWpb",
          "status": "ltWk5OffTrack",
          "submitterFirstName": "Ephraim",
          "submitterLastName": "Malinga",
          "weeksSpent": "N/A",
        }
      }
      fellowCells={
        Array [
          Object {
            "color": "no-track",
            "element": "name",
            "value": "N/A",
          },
          Object {
            "color": null,
            "element": "level",
            "value": "D0A Simulations",
          },
          Object {
            "color": "no-track",
            "element": "weeksSpent",
            "value": "N/A",
          },
          Object {
            "color": "no-track",
            "element": "ttlName",
            "value": "N/A",
          },
          Object {
            "color": "off-track",
            "element": "devPulseStatus",
            "value": "Off Track",
          },
          Object {
            "color": "no-track",
            "element": "lmsStatus",
            "value": "N/A",
          },
          Object {
            "color": "no-track",
            "element": "advanceStatus",
            "value": "N/A",
          },
        ]
      }
      key="-LJK2KDhq70_X1YbirFa"
    />
  </Table>
</Fragment>
`;

exports[`Dashboard Table  renders the sort buttons to match snapshot 1`] = `
<span
  className="sort-buttons ml-3 p-0 text-left"
>
  <div
    className="sort-buttons__up p-0"
  >
    <i
      className="fas align-self-center fa-angle-up text-primary"
      data-ascending={true}
      data-target="header"
      onClick={[MockFunction]}
      onKeyPress={[MockFunction]}
      role="button"
      tabIndex="0"
    />
  </div>
  <div
    className="sort-buttons__down p-0"
  >
    <i
      className="fas align-self-center fa-angle-down "
      data-target="header"
      onClick={[MockFunction]}
      onKeyPress={[MockFunction]}
      role="button"
      tabIndex="-1"
    />
  </div>
</span>
`;
